modules/api/src/main/java/net/countercraft/movecraft/MovecraftRepair.java

```diff
@@ -477,4 +486,68 @@ public static void initialize(Plugin plugin){
     public static MovecraftRepair getInstance() {
         return instance;
     }
+
+    public int convertOldCraftRepairStates(){
+        //Check in the old RepairStates folder
+        File repairStateDir = new File(plugin.getDataFolder(),"RepairStates");
+        if (!repairStateDir.exists()){
+            return 0;
+        }
+        int convertedRepairStates = 0;
+        File[] repairStates = repairStateDir.listFiles();
+        if (repairStateDir.exists() && (repairStates != null || repairStates.length > 0)){
+            for (File rs : repairStates){
+                if (!rs.getName().contains(".schematic")){
+                    continue;
+                }
+                String fileName = rs.getName();
+                fileName = fileName.replace(".schematic", "");
+                OfflinePlayer owner = null;
+                if (fileName.contains("_")) {
+                    String[] parts = fileName.split("_");
```

Comment:
underscores can exist in usernames

Reply:
You should probably generate a map from Strings to UUID's using `Bukkit::getOfflinePlayers`, than , in this section , iterate  through the keys of the map to check if there are any names for which `fileName.substring(key)` returns true, add them to another list. If this list has size greater than 1, you'll need to skip the file and log an error. otherwise, the first name is the correct one.